apt/risk/data_assessment/dataset_attack_membership_classification.py

```diff
@@ -0,0 +1,160 @@
+    """Configuration for DatasetAttackMembershipClassification.
+
+    Attributes:
+        classifier_type:  sklearn classifier type for the member classification.
+                          Can be LogisticRegression or RandomForestClassifier
+        threshold:  a minimum threshold of distinguishability, above which a synthetic_data_quality_warning is raised.
+                    A value higher than the threshold means that it is too easy to distinguish between the synthetic
+                    data and the training or test data.
+    """
+    classifier_type: str = 'RandomForestClassifier'
+    threshold: float = 0.9
+
+
+@dataclass
+class DatasetAttackScoreMembershipClassification(DatasetAttackScore):
+    """DatasetAttackMembershipClassification privacy risk score.
+    """
+    member_roc_auc_score: float
+    non_member_roc_auc_score: float
+    normalized_ratio: float
+    synthetic_data_quality_warning: bool
+    assessment_type: str = 'MembershipClassification'  # to be used in reports
+
+    def __init__(self, dataset_name: str, member_roc_auc_score: float, non_member_roc_auc_score: float,
+                 normalized_ratio: float, synthetic_data_quality_warning: bool) -> None:
+        """
+        dataset_name:    dataset name to be used in reports
+        member_roc_auc_score:  ROC AUC score of classification between members (training) data and synthetic data
+        non_member_roc_auc_score: ROC AUC score of classification between non-members (test) data and synthetic data,
+                                    this is the baseline score
+        normalized_ratio:  ratio of the member_roc_auc_score to the non_member_roc_auc_score
+        synthetic_data_quality_warning:  True if either the member_roc_auc_score or the non_member_roc_auc_score is
+                                        higher than the threshold. That means that the synthetic data does not represent
+                                        the training data sufficiently well, or that the test data is too far from the
+                                        synthetic data.
+        """
+        super().__init__(dataset_name=dataset_name, risk_score=normalized_ratio, result=None)
+        self.member_roc_auc_score = member_roc_auc_score
+        self.non_member_roc_auc_score = non_member_roc_auc_score
+        self.normalized_ratio = normalized_ratio
+        self.synthetic_data_quality_warning = synthetic_data_quality_warning
+
+
+class DatasetAttackMembershipClassification(DatasetAttackMembership):
+    """
+         Privacy risk assessment for synthetic datasets that compares the distinguishability of the synthetic dataset
+         from the members dataset (training) as opposed to the distinguishability of the synthetic dataset from the
+         non-members dataset (test).
+         The privacy risk measure is calculated as the ratio of the receiver operating characteristic curve (AUC ROC) of
+         the members dataset to AU ROC of the non-members dataset.
```

Comment:
missing C (AUC)

Reply:
Also, maybe warrants a short explanation of the resulting score (i.e, this score is above 0, unbounded, the higher the score the worse the privacy).